checkov/terraform/checks/resource/aws/CloudfrontDistributionEncryption.py

```diff
@@ -23,8 +23,10 @@ def scan_resource_conf(self, conf):
                 return CheckResult.FAILED
         if "ordered_cache_behavior" in conf.keys():
             for behavior in conf["ordered_cache_behavior"]:
-                if behavior["viewer_protocol_policy"][0] == "allow-all":
-                    return CheckResult.FAILED
+                if type(behavior) is dict:
```

Comment:
can you give an example of a non-dict value in tf file?

Reply:
yes, ther error happened in a tf file in which the ordered_cache behavior is written as a dynamic, so the behavior was a string - ```'${var.ordered_cache}'```. 
this is the tf file, I also added it to the example.tf in checkov (part of this commit):

```
dynamic "ordered_cache_behavior" {
    for_each = var.ordered_cache

    content {
      path_pattern = ordered_cache_behavior.value.path_pattern

      allowed_methods  = ordered_cache_behavior.value.allowed_methods
      cached_methods   = ordered_cache_behavior.value.cached_methods
      target_origin_id = module.distribution_label.id
      compress         = ordered_cache_behavior.value.compress
      trusted_signers  = var.trusted_signers

      forwarded_values {
        query_string = ordered_cache_behavior.value.forward_query_string
        headers      = ordered_cache_behavior.value.forward_header_values

        cookies {
          forward = ordered_cache_behavior.value.forward_cookies
        }
      }

      viewer_protocol_policy = ordered_cache_behavior.value.viewer_protocol_policy
      default_ttl            = ordered_cache_behavior.value.default_ttl
      min_ttl                = ordered_cache_behavior.value.min_ttl
      max_ttl                = ordered_cache_behavior.value.max_ttl

      dynamic "lambda_function_association" {
        for_each = ordered_cache_behavior.value.lambda_function_association
        content {
          event_type   = lambda_function_association.value.event_type
          include_body = lookup(lambda_function_association.value, "include_body", null)
          lambda_arn   = lambda_function_association.value.lambda_arn
        }
      }
    }
  }
```

Reply:
@arielkru The example above is an example of a dict, not of a string

Reply:
@nimrodkor checkov gets this: '${var.ordered_cache}' and raises an error for being a string and not dict.. I think its because the variable ```'ordered_cache'``` in the variables file is set to ```[]```, so maybe the ```for_each = var.ordered_cache``` gets nothing.
Im not sure anyway, you can check out how checkov reads this file in the example.tf